tests/test_annotated.py

```diff
@@ -76,3 +76,35 @@ def cmd(force: Annotated[bool, typer.Option("--force")] = False):
     result = runner.invoke(app, ["--force"])
     assert result.exit_code == 0, result.output
     assert "Forcing operation" in result.output
+
+
+def test_default_with_class_with_custom_eq():
```

Comment:
Tiangolo: we might want this test in a different location in the test suite?

Reply:
Good point! I just moved it to another file. Not sure about the name, but it will work for now. :shrug: